tests/transformers/test_transformers_llm_inference_utils.py

```diff
@@ -61,40 +62,131 @@ def apply_chat_template(self, messages: List[Dict[str, str]], **kwargs):
 
 
 def test_apply_chat_template():
-    tokenizer = DummyTokenizer()
-
-    data1 = pd.DataFrame(
-        {
-            "messages": pd.Series(
-                [[{"role": "A", "content": "one"}, {"role": "B", "content": "two"}]]
-            ),
-            "random": ["value"],
-        }
-    )
-
+    data1 = [{"role": "A", "content": "one"}, {"role": "B", "content": "two"}]
     # Test that the function modifies the data in place for Chat task
-    convert_data_messages_with_chat_template(data1, tokenizer)
+    prompt = convert_data_messages_with_chat_template(data1, DummyTokenizer())
+    assert prompt == ["one two"]
 
-    expected_data = pd.DataFrame({"random": ["value"], "prompt": ["one two"]})
-    pd.testing.assert_frame_equal(data1, expected_data)
 
+_TestCase = namedtuple("_TestCase", ["data", "params", "expected_data", "expected_params"])
 
-def test_preprocess_llm_inference_params():
-    data = pd.DataFrame(
-        {
-            "prompt": ["Hello world!"],
-            "temperature": [0.7],
-            "max_tokens": [100],
-            # do not pass this to params as it is None
-            "stop": None,
-        }
-    )
 
-    data, params = preprocess_llm_inference_params(data, flavor_config=None)
+@pytest.mark.parametrize(
+    "case",
+    [
+        # Case 0: Data only includes prompt
+        _TestCase(
+            data={"prompt": ["Hello world!"]},
+            params={},
+            expected_data=["Hello world!"],
+            expected_params={},
+        ),
+        # Case 1: Data includes prompt and params
+        _TestCase(
+            data={
+                "prompt": ["Hello world!"],
+                "temperature": [0.7],
+                "max_tokens": [100],
+                "stop": None,
+            },
+            params={},
+            expected_data=["Hello world!"],
+            expected_params={
+                "temperature": 0.7,
+                # max_tokens is replaced with max_new_tokens
+                "max_new_tokens": 100,
+                # do not pass `stop` to params as it is None
```

Comment:
Is the behavior identical if 
```python
"stop": [],
``` 
?

Reply:
it has to be.. lemme check just in case

Reply:
Ah wait it's not same. `"stop": []` overrides model config to not to stop with any word, while `"stop": None"` is no-op (use model config). @BenWilson2 do you think it's ok? My feeling is it's ok cuz it aligns with how we consider empty-vs-null in general, but may be confusing sometimes.